### PR TITLE
Use proxy when configured for the base repo (#1196953)

### PR DESCRIPTION
**Pull request by spartacus06**
Signed-off-by: Seth Jennings <sjenning@redhat.com>

### CODE REVIEW

File: pyanaconda/packaging/dnfpayload.py [hunk 1]
```diff
@@ -227,6 +227,12 @@ def _add_repo(self, ksrepo):
         if mirrorlist:
             repo.mirrorlist = mirrorlist
         repo.sslverify = not (ksrepo.noverifyssl or flags.noverifyssl)
+        if ksrepo.proxy:
+                try:
+                    repo.proxy = ProxyString(ksrepo.proxy).url
+                except ProxyStringError as e:
+                    log.error("Failed to parse proxy for _add_repo %s: %s",
+                              ksrepo.proxy, e)
```
Comment (vpodzime) [inline pyanaconda/packaging/dnfpayload.py]: The indentation of this block is ugly, maybe even wrong.

File: pyanaconda/packaging/dnfpayload.py [hunk 2]
```diff
@@ -702,9 +708,13 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
                           method.method, e)
 
             try:
+                if hasattr(method, "proxy"):
+                    proxy = method.proxy
+                else:
+                    proxy = None
```
Comment (vpodzime) [inline pyanaconda/packaging/dnfpayload.py]: You can use the ``getattr()`` function with ``None`` used as the default value (second argument) here.

**Comment by vpodzime** [comment]
First of all, thanks for the patch! Could you please fix the two minor issues reported above and push-force a new version of the commit to you branch (IOW, update the pull request)?

**Comment by spartacus06** [comment]
PR updated with requested changes

File: pyanaconda/packaging/dnfpayload.py [hunk 3]
```diff
@@ -702,9 +708,10 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
                           method.method, e)
 
             try:
+                proxy = getattr(method, "proxy", None);
```
Comment (vpodzime) [inline pyanaconda/packaging/dnfpayload.py]: No semicolons in python, please. ;)

**Comment by vpodzime** [comment]
Apart from the last minor thing above (that can be fixed as part of pushing this change) this looks good to me.

File: pyanaconda/packaging/dnfpayload.py (hunk 3 above)
Comment (spartacus06) [inline pyanaconda/packaging/dnfpayload.py]: doh!

**Comment by spartacus06** [comment]
I went ahead and fixed it up myself.  Thanks for the review and ack @vpodzime

**Comment by vpodzime** [comment]
Pushed some time ago.

**Pull request closed without merging**